File: application/src/main/java/ru/artemev/application/dto/LoanOfferDTO.java
package ru.artemev.application.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Builder;
import lombok.Data;

import java.math.BigDecimal;

@Data
@Builder
@Schema(description = "Предложение по кредиту")
@JsonDeserialize(builder = LoanOfferDTO.LoanOfferDTOBuilder.class)
public class LoanOfferDTO {
  @Schema(description = "Итоговая сумма", example = "15000")
  @JsonProperty("totalAmount")
  private final BigDecimal totalAmount;

  @Schema(description = "Срок займа", example = "10")
  @JsonProperty("term")
  private final Integer term;

  @Schema(description = "Ежемесячный платеж", example = "1500")
  @JsonProperty("monthlyPayment")
  private final BigDecimal monthlyPayment;

  @Schema(description = "Процентная ставка", example = "10")
  @JsonProperty("rate")
  private final BigDecimal rate;

  @Schema(description = "Страховка", example = "true")
  @JsonProperty("isInsuranceEnabled")
  private final Boolean isInsuranceEnabled;

  @Schema(description = "Зарплатный клиент", example = "true")
  @JsonProperty("isSalaryClient")
  private final Boolean isSalaryClient;

  @Schema(description = "Номер предложения", example = "4")
  @JsonProperty("applicationId")
  private Long applicationId;

  @Schema(description = "Запрашиваемая сумма", example = "100000")
  @JsonProperty("requestedAmount")
  private BigDecimal requestedAmount;
}

File: gateway/src/main/kotlin/ru/artemev/gateway/dto/LoanOfferDTO.kt
package ru.artemev.gateway.dto

import com.fasterxml.jackson.annotation.JsonProperty
import io.swagger.v3.oas.annotations.media.Schema
import java.math.BigDecimal

@Schema(description = "Предложение по кредиту")
data class LoanOfferDTO(
    @Schema(description = "Итоговая сумма", example = "15000")
    @JsonProperty("totalAmount")
    val totalAmount: BigDecimal? = null,

    @Schema(description = "Срок займа", example = "10")
    @JsonProperty("term")
    val term: Int? = null,

    @Schema(description = "Ежемесячный платеж", example = "1500")
    @JsonProperty("monthlyPayment")
    val monthlyPayment: BigDecimal? = null,

    @Schema(description = "Процентная ставка", example = "10")
    @JsonProperty("rate")
    val rate: BigDecimal? = null,

    @Schema(description = "Страховка", example = "true")
    @JsonProperty("isInsuranceEnabled")
    val isInsuranceEnabled: Boolean = false,

    @Schema(description = "Зарплатный клиент", example = "true")
    @JsonProperty("isSalaryClient")
    val isSalaryClient: Boolean = false,

    @Schema(description = "Номер предложения", example = "4")
    @JsonProperty("applicationId")
    var applicationId: Long? = null,

    @Schema(description = "Запрашиваемая сумма", example = "100000")
    @JsonProperty("requestedAmount")
    val requestedAmount: BigDecimal? = null
)
